### PR TITLE
Fix GraphQL Resolvers not Passing Enum Values Correctly when Used as Input Parameters

### PR DESCRIPTION
**Pull request by ThisaruGuruge**
## Purpose

$Subject

Fixes: [#3721](https://github.com/ballerina-platform/ballerina-standard-library/issues/3721)

## Checklist
- [x] Linked to an issue
- [x] Updated the changelog
- [x] Added tests
- [ ] ~Updated the spec~
- [x] Checked native-image compatibility

### CODE REVIEW

File: ballerina-tests/tests/test_services.bal [hunk 1]
```diff
@@ -742,6 +742,34 @@ service /special_types on specialTypesTestListener {
     resource function get company() returns Company {
         return company;
     }
+
+    resource function get month(Month month) returns string {
```
Comment (MohamedSabthar) [inline ballerina-tests/tests/test_services.bal:746]: Isn't it necessary to return the Month in order to verify the bug fix?

Comment (ThisaruGuruge) [reply ballerina-tests/tests/test_services.bal:746]: Yes, otherwise the passed string would be `JANUARY`, `February`, etc.

Comment (DimuthuMadushan) [reply ballerina-tests/tests/test_services.bal:746]: If we return the `Month` type here(just like the example in the issue), we can verify the output value as well.

Comment (ThisaruGuruge) [reply ballerina-tests/tests/test_services.bal:746]: Right. I think you are correct. Will update the code.

Comment (ThisaruGuruge) [reply ballerina-tests/tests/test_services.bal:746]: Fixed with 008f37bb942e035068628ce8a344c0ff201ddfe7

**Review by DimuthuMadushan — APPROVED** [review]

**Review by MohamedSabthar — APPROVED** [review]

**Pull request merged**